internal/service/verify/data_source_verify_policy_test.go

```diff
@@ -0,0 +1,346 @@
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheckEnvironment(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckVerifyPolicyDestroy,
+		ErrorCheck:               acctest.ErrorCheck(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVerifyPolicy_FindByID(environmentName, licenseID, resourceName, name),
+				Check:  findByID,
+			},
+			{
+				Config:  testAccVerifyPolicy_FindByID(environmentName, licenseID, resourceName, name),
+				Destroy: true,
+			},
+			{
+				Config: testAccVerifyPolicy_FindByName(environmentName, licenseID, resourceName, updatedName),
+				Check:  findByName,
+			},
+			{
+				Config:  testAccVerifyPolicy_FindByName(environmentName, licenseID, resourceName, updatedName),
+				Destroy: true,
+			},
+			{
+				Config: testAccVerifyPolicy_FindDefaultPolicy(environmentName, licenseID, resourceName, updatedName),
+				Check:  findDefault,
+			},
+		},
+	})
+}
+
+func TestAccVerifyPolicyDataSource_FailureChecks(t *testing.T) {
+	t.Parallel()
+
+	resourceName := acctest.ResourceNameGen()
+
+	name := acctest.ResourceNameGen()
+
+	environmentName := acctest.ResourceNameGenEnvironment()
+	licenseID := os.Getenv("PINGONE_LICENSE_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheckEnvironment(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckVerifyPolicyDestroy,
+		ErrorCheck:               acctest.ErrorCheck(t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccVerifyPolicy_FindByIDFail(environmentName, licenseID, resourceName, name),
+				ExpectError: regexp.MustCompile("Error: Error when calling `ReadOneVerifyPolicy`: 404 Not Found"),
+			},
+			{
+				Config:      testAccVerifyPolicy_FindByNameFail(environmentName, licenseID, resourceName, name),
+				ExpectError: regexp.MustCompile("Error: Cannot find verify policy from name"),
+			},
+		},
+	})
+}
+
+func testAccVerifyPolicy_FindByID(environmentName, licenseID, resourceName, name string) string {
+	return fmt.Sprintf(`
+	%[1]s
+resource "pingone_verify_policy" "%[3]s" {
+  environment_id = pingone_environment.%[2]s.id
+  name           = "%[4]s"
+  description    = "%[4]s"
+  default        = false
+
+  government_id = {
+    verify = "REQUIRED"
+  }
+
+  facial_comparison = {
+    verify    = "REQUIRED"
+    threshold = "HIGH"
+  }
+
+  liveness = {
+    verify    = "REQUIRED"
+    threshold = "HIGH"
+  }
+
+  email = {
+    verify = "REQUIRED"
+    create_mfa_device : true
+    otp = {
+      attempts = {
+        count = "4"
+      }
+      lifetime = {
+        duration  = "16"
+        time_unit = "MINUTES"
+      },
+      deliveries = {
+        count = 5
+        cooldown = {
+          duration  = "33"
+          time_unit = "SECONDS"
+        }
+      }
+      notification = {
+        variant_name = "english_b"
```

Comment:
The `variant_name` _should_ align to a notification template variant name, however the Verify Policy API does not check to see if the `variant_name` exists, so I provided the same behavior in the provider (for better or worse).

I've gone back-and-forth on this (and similar items).  But in this case, I will add a check to ensure that the variant name is valid/exists.

Reply:
Added validation checks.  Currently regression testing.

Reply:
Added function, but without support for the p1verify notification template (see [Issue416](https://github.com/pingidentity/terraform-provider-pingone/issues/416)), implementing proper test cases is not possible without disabling the enforcement on the notification template name requirements in the verify service.

Recommend we move forward with the current PR as-is, which mimics the same functionality as the API currently.  Then follow-up with the notification template enhancements for the base service, enabling proper checks/testing in p1verify service.

Reply:
We can move forward as it is.  Sounds like a JIRA is required to clarify the proper behaviour of the API in checking the variant existence - the only reason we would add these checks (which imo are optional as they should be caught by the API) is to provide user friendly output prior to the `POST` call, rather than do the job of the API